executor/aggfuncs/func_sum.go

```diff
@@ -38,7 +39,7 @@ type partialResult4SumDistinctFloat64 struct {
 
 type partialResult4SumDistinctDecimal struct {
 	partialResult4SumDecimal
-	valSet set.DecimalSet
```

Comment:
How about keep `DecimalSet` and use `MyDecimal.ToHashKey` to make `DecimalSet.Exist` correct.

Reply:
If we keep DecimalSet here, we'll call ToHashKey twice if `DecimalSet.Exist()` returns false. One in `DecimalSet.Exist()`, another in `DecimalSet.Insert()`. @qw4990

Reply:
Maybe we can add one method like `InsertIfNotExists` and return whether it's a value to be inserted.

One question is that, is the behavior of the current `DecimalSet` right?

Reply:
> One question is that, is the behavior of the current DecimalSet right?

Do you find anything wrong? @winoros

Reply:
No, since the `DecimalSet` only be in used in the places you changed in this pr.

I just wonder that whether there'll be some cases in the future that will need the original `DecimalSet`.

Reply:
DecimalSet will not be used by other places now. It was introduced when we were refactoring the agg. @winoros